src/thin_tree.rs

```diff
@@ -0,0 +1,68 @@
+use alloc::collections::btree_map::BTreeMap;
+
+use crate::tar_parser::{Parser, ParserError};
+use crate::StrFilename;
+
+#[derive(Clone, Debug, PartialEq, Eq, yoke::Yokeable)]
+pub enum ThinTreeRef<'a> {
+    File(&'a [u8]),
+    Directory(BTreeMap<&'a str, ThinTreeRef<'a>>),
+}
+
+impl<'a> ThinTreeRef<'a> {
+    /// Populate a thin directory tree, with `entry` pointing to `r`
+    pub fn update(&mut self, entry: StrFilename<'a>, r: &'a [u8]) -> Result<(), ParserError<'a>> {
```

Comment:
does this need to be `pub`?

Reply:
(Same for the other functions here)

Reply:
I'd take a conservative approach here. Everything that is pub might lead to hazzle when for some reason we have to remove it later on. We can still make it pub if we find a use-case.

Reply:
For `resolve` and `resolve_mut` the use case is obvious.